conans/client/source.py

```diff
@@ -81,8 +82,6 @@ def is_excluded(origin_path):
             else:
                 shutil.copy2(src_file, dst_file)
 
-    FileCopier.link_folders(src, dst, linked_folders)
```

Comment:
I barely remember this but I would say the links were done at the end for something, maybe when the dest dir doesn't exist yet? or when it should point to the package? I can't remember

Reply:
The `FileCopier.link_folders` is for example removing "empty links" and converting to relative... are we sure we are not going to introduce another regression?

Reply:
That was the idea, but we don't need to wait until everything is copied because we can create symlinks to files/folders that aren't already there (it works with files, as seen L81, and it works with folders too).

And here we were introducing a bug: we were creating links from the `source` folder to `export_source` one, instead of creating links to the folders that have been copied.

Reply:
And the "broken" ones? we kept them broken?

Reply:
Broken symlinks should be noticed before copying from one folder of the cache to another folder (probably during the `export` or the `unzip` stages). I've just tested and a broken symlink is being detected here: https://github.com/conan-io/conan/blob/46663964cbe546f50d29901ce749d694bba117c9/conans/model/manifest.py#L41

Reply:
I've added a test to the `develop` branch, it's better to be sure: https://github.com/conan-io/conan/pull/5345